scipy/sparse/base.py

```diff
@@ -386,6 +386,16 @@ def __rmul__(self, other):  # other * self
                 tr = np.asarray(other).transpose()
             return (self.transpose() * tr).transpose()
 
+    #####################################
+    # matmul (@) operator (Python 3.5+) #
+    #####################################
+
+    def __matmul__(self, other):
+        return self * other
```

Comment:
Probably better as `self.__mul__(other)` so that NotImplemented is in correct layer.